### PR TITLE
Add draft of the SDK Rustdoc

### LINKED ISSUE #20: Document public interfaces for `rustdoc`
The rustdocs can be generated with `cargo doc --no-deps --open`.

This will build the docs locally and open them in a browser.

### PR DESCRIPTION
**Pull request by ok300**
The rustdocs can be generated with `cargo doc --no-deps --open`. This will build the docs locally and open them in a browser.

Fixes #20

### CODE REVIEW

File: libs/sdk-core/src/lib.rs [hunk 1]
```diff
@@ -1,3 +1,143 @@
...
+//! Supported BOLT11 operations are
+//!
+//! * [BreezServices::receive_payment] to create an invoice
+//! * [BreezServices::send_payment] to pay an invoice
+//!
+//! Keysend is possible with [BreezServices::send_spontaneous_payment].
+//!
+//! See LNURL section below for LNURL-pay and -withdraw.
+//!
+//! ### C. On-chain Operations
```
Comment (roeierez) [inline libs/sdk-core/src/lib.rs:90]: I think it is better to have a category for On-chain Receive (swap in) with the following methods:
[BreezServices::receive_onchain]
[BreezServices::list_refundables] for a list of swaps
[BreezServices::refund]

File: libs/sdk-core/src/models.rs [hunk 2]
```diff
@@ -271,13 +287,15 @@ pub struct LnPaymentDetails {
...
+/// Details for a payment which closed a channel
 #[derive(PartialEq, Eq, Debug, Clone, Deserialize, Serialize)]
-pub struct ClosesChannelPaymentDetails {
+pub struct ClosedChannelPaymentDetails {
     pub short_channel_id: String,
     pub state: ChannelState,
     pub funding_txid: String,
 }
 
+/// LN channel managed by the LSP
```
Comment (roeierez) [inline libs/sdk-core/src/models.rs:298]: Not managed and doesn't have to be with the LSP peer. I would use just an "LN channel"

File: libs/sdk-core/src/lib.rs [hunk 3]
```diff
@@ -1,3 +1,143 @@
...
+//! ```ignore
+//! let mnemonic = Mnemonic::new(Words12, English);
+//! let seed = Seed::new(&mnemonic, "");
+//!
+//! let creds = BreezServices::register_node(Network::Bitcoin, seed.as_bytes().to_vec()).await?;
+//! let sdk = BreezServices::init_services(
+//!         None,
+//!         seed.to_vec(),
+//!         creds.clone(),
+//!         Box::new(AppEventListener {}),
```
Comment (roeierez) [inline libs/sdk-core/src/lib.rs:31]: I think in the following sections where you put some great details about the possibilities it worth adding a section for the event listener and which events can be consumed by the user.

**Review by roeierez — CHANGES_REQUESTED** [review]
Some minor comments and suggestions. Looks great!

File: libs/sdk-core/src/lib.rs (hunk 3 above)
Comment (ok300) [reply libs/sdk-core/src/lib.rs:31]: Done, I added a paragraph with links in the intro section.

File: libs/sdk-core/src/lib.rs (hunk 1 above)
Comment (ok300) [reply libs/sdk-core/src/lib.rs]: Done. I then moved `[BreezServices::sweep]` to the Wallet Utilities section.

File: libs/sdk-core/src/models.rs (hunk 2 above)
Comment (ok300) [reply libs/sdk-core/src/models.rs]: Done

File: libs/sdk-core/src/models.rs [hunk 4]
```diff
@@ -271,13 +287,15 @@ pub struct LnPaymentDetails {
     pub bolt11: String,
 }
 
+/// Details for a payment which closed a channel
```
[the remark this answers is not in the document]
Comment (ok300) [reply libs/sdk-core/src/models.rs]: Done. Not very familiar with the inner workings of this, so if there's a clearer better way to describe it, let me know.

**Review by roeierez — APPROVED** [review]

**Pull request merged**